### DISCUSSION TITLE
Linking model by generic association table use case

### DISCUSSION THREAD

**Discussion by ofekisr** [sqlalchemy/sqlalchemy#7188] [Usage Questions] [answered]
Flask-SQLAlchemy 2.5.1  
SQLAlchemy       1.3.24  
SQLAlchemy-Utils 0.36.8

I'm struggling with linking a Regular Model by a generic association table. 

There is a regular Model - Dashboard,  
A Role model who has only id and name columns  
A generic association table (ObjectRoles) that link any regular model to Role model - ManyToMany

```python
from flask_appbuilder import Model
# the Model inherit from DefaultMeta of flask-sqlalchemy
from sqlalchemy import Column, ForeignKey, Integer, String
from sqlalchemy.orm import relationship
from sqlalchemy_utils import generic_relationship


class ObjectRoles(Model):  
    __tablename__ = "object_roles"
    id = Column(Integer, primary_key=True)
    role_id = Column("role_id", Integer, ForeignKey("ab_role.id"), nullable=False)
    object_id = Column(Integer, nullable=False)   
    object_type = Column(String(255), nullable=False)
    object = generic_relationship(object_type, object_id)  # not sure if is required 
    role = relationship(Role)                              # not sure if is required 


class Dashboard(Model):
    __tablename__ = "dashboards"
    id = Column(Integer, primary_key=True)
    roles = # Don't know what to do here, I've tried many alternatives

    # roles = relationship(ObjectRoles,
    #                       primaryjoin="and_(Dashboard.id==ObjectRoles.object_id, "
    #                                   "ObjectRoles.object_type=='Dashboard')",
    #                       )


```
I need the ability to fetch the roles like the primary join and able to insert a new roles to the dashboard, so behind the scene it will create a new row in object_roles

if the association table wasn't generic but with foreign key to Dashaboard 
there was not problems

I post the question in [StackOverflow](https://stackoverflow.com/questions/69568619/sqlalchemy-and-flask-sqlalchemy-generic-association-table) too

**Comment by zzzeek** [comment, marked as answer]
hi there -

this looks to be a sqlalchemy_utils question.  We don't support sqlalchemy_utils here so you likely want to ask at https://github.com/kvesteri/sqlalchemy-utils .

Within plain SQLAlchemy, so-called "generic" relationships are illustrated directly at https://docs.sqlalchemy.org/en/14/orm/examples.html#module-examples.generic_associations

**Reply by ofekisr** [reply]
Thanks, 
after reading the blog and the examples I figured out how to construct generic many2many associations :)
